Khan/perseus
# Publish npm snapshot on pull request create/update

## Summary:

This PR introduces a new Github Action job to publish an npm snapshot each time a PR is created/updated. The intent is to make it as easy as possible to test Perseus changes in other projects. 

The snapshot publish follows a very similar path to the `publish:ci` yarn script except that it publishes the packages under a tag named with the pattern `PR${pull_reqeust_number` (ie. `PR522`). This allows us to easily install the published packages by installing them using a command such as `yarn add @khanacademy/perseus@PR522`. 

### npm tags 

> Tags can be used to provide an alias instead of version numbers.
> 
> For example, a project might choose to have multiple streams of development and use a different tag for each stream, e.g., stable, beta, dev, canary.
> 
> By default, the latest tag is used by npm to identify the current version of a package, and npm install <pkg> (without any @<version> or @<tag> specifier) installs the latest tag. Typically, projects only use the latest tag for stable release versions, and use other tags for unstable versions such as prereleases.
> 
> The next tag is used by some projects to identify the upcoming version.
> 
> Other than latest, no tag has any special significance to npm itself.

### Commenting 

This action will comment on the PR if it is able to publish a snapshot or not. I'm not able to check the success comment here because we currently don't have any change sets on `main`.  I'll keep an eye on this as other PRs land. Here's the failure comment though. 

<img width="450" alt="image" src="https://user-images.githubusercontent.com/77138/225444144-7821be8f-c1ba-4602-bfa1-dc07848d6d4e.png">


https://docs.npmjs.com/cli/v9/commands/npm-dist-tag#purpose

Issue: LC-590

## Test plan:

- Land this PR
- Start a new PR and make some changes to Perseus (and one or two other packages in this repo). 
- Open a new PR
- Wait until the new `publish_snapshot` job has completed and verify it did the right thing(s)
- Update the PR with new changes
- Ensure that the `publish_snapshot` job re-published and overwrote the existing npm tag
